lib/ddtrace/pipeline.rb

```diff
@@ -9,6 +9,14 @@ module Pipeline
     @mutex = Mutex.new
     @processors = []
 
+    # {.before_flush} allows application to alter or filter out traces before they are flushed.
+    #
+    # @see file:docs/ProcessingPipeline.md Processing Pipeline
+    # @param [Array<Proc>] processors a list of callable objects that receive a list of {Datadog::Span}s and can modify
+    #   or filter our spans.
+    # @yield Optional that receives an array of spans and returns the desired remaining spans.
```

Comment:
I'm not sure I would understand what this means if I hadn't looked at the examples in the docs. Furthermore, it's not clear anywhere that if users provide both a list of processors as well as a block, the block gets ignored (should we instead raise an exception when that happens?).

Reply:
I agree with it.

Trying to document this method I can see that the API is overloaded, which makes documentation hard as well.

It's very convenient to allow both a callable object as well as a block, but it makes it hard to explain the nuance.

I've took a shot and rewrote most of these.